components/feature/media/src/main/java/mozilla/components/feature/media/service/MediaSessionServiceDelegate.kt

```diff
@@ -127,38 +119,57 @@ internal class MediaSessionServiceDelegate(
         shutdown()
     }
 
-    private fun processMediaSessionState(state: SessionState?) {
-        if (state == null) {
-            shutdown()
-            return
-        }
+    override fun handleMediaPlaying(sessionState: SessionState) {
+        emitStatePlayFact()
 
-        when (state.mediaSessionState?.playbackState) {
-            MediaSession.PlaybackState.PLAYING -> {
-                registerBecomingNoisyListenerIfNeeded(state)
-                audioFocus.request(state.id)
-                emitStatePlayFact()
-                startForegroundNotificationIfNeeded()
-            }
-            MediaSession.PlaybackState.PAUSED -> {
-                unregisterBecomingNoisyListenerIfNeeded()
-                emitStatePauseFact()
-                stopForeground()
-            }
-            else -> {
-                unregisterBecomingNoisyListenerIfNeeded()
-                emitStateStopFact()
-                stopForeground()
+        updateMediaSession(sessionState)
+        registerBecomingNoisyListenerIfNeeded(sessionState)
+        audioFocus.request(sessionState.id)
+        controller = sessionState.mediaSessionState?.controller
+
+        notificationScope?.launch {
+            val notification = notification.create(sessionState, mediaSession)
+            if (isForegroundService) {
+                getNotificationManager().notify(notificationId, notification)
+            } else {
+                service.startForeground(notificationId, notification)
+                isForegroundService = true
             }
```

Comment:
The name may be a bit misleading since we have an if/else and will not everytime "startForeground".
Can keep the if/else here but refactor these lines to two new methods, with the one for just posting a notification then used for the other "handle.." methods also.